ambry-clustermap/src/main/java/com/github/ambry/clustermap/DatacenterInitializer.java

```diff
@@ -197,7 +200,8 @@ private DcInfo initializeHelixDatacenter() throws Exception {
     // notification for a change from within the same thread that adds the listener, in the context of the add
     // call. Therefore, when the call to add a listener returns, the initial notification will have been
     // received and handled.
-    DataNodeConfigSource dataNodeConfigSource = new InstanceConfigToDataNodeConfigAdapter(manager, clusterMapConfig);
+    DataNodeConfigSource dataNodeConfigSource =
+        ClusterMapUtils.getDataNodeConfigSource(clusterMapConfig, manager, dcName, dataNodeConfigSourceMetrics);
```

Comment:
I felt that it was extra data that did not need to be in each znode since it can be inferred from the zookeeper connected to. If you think that it is better to have it there (for example, if we perform multi-datacenter aggregation in the future), I can definitely add it back in. What do you think?

Reply:
I see your point but I feel like adding `DATACENTER_STR` field won't bring too much overhead and in the long term I believe multi-datacenter aggregation is needed. I see two benefits of aggregation view:  1. reduce cross-colo read qps on ZK. 2. speed up startup time for both fronend/server (don't need to wait for remote dc initial notification). Adding `DATACENTER_STR` back is helpful in future case.

Reply:
Sure, that definitely makes sense so that we won't have  to make an incremental schema change in the future just for this. My most recent commit has added the datacenter field back.